md/moves.py

```diff
@@ -56,3 +56,45 @@ def n_accepted(self):
     @property
     def n_proposed(self):
         return np.sum(self.n_proposed_by_move)
+
+
+from scipy.special import logsumexp
+
+
+class MultipleTryMetropolis:
+    def __init__(self, K):
+        self.K = K
+        self.n_accepted = 0
+        self.n_proposed = 0
+
+    def batch_propose(self, x: CoordsVelBox):
+        raise NotImplementedError()
+
+    def batch_log_prob(self, proposals):
+        raise NotImplementedError()
+
+    def move(self, x: CoordsVelBox) -> CoordsVelBox:
+        self.n_proposed += 1
+        # assume proposals themselves have water and ligand aligned and combined
+        numerator_proposals = self.batch_propose(x)
+        numerator_log_probs = self.batch_log_prob(numerator_proposals)
+        assert self.K == len(numerator_proposals)
+        assert self.K == len(numerator_log_probs)
+        normalized_numerator_log_probs = np.exp(numerator_log_probs - logsumexp(numerator_log_probs))
+
+        new_sample = numerator_proposals[np.random.choice(np.arange(self.K), p=normalized_numerator_log_probs)]
```

Comment:
naming nit: "new_sample" -> "x_proposed", to be parallel to "x"

Reply:
willfix